jpscore/src/main.cpp

```diff
@@ -74,41 +73,28 @@ int main(int argc, char ** argv)
         return EXIT_FAILURE;
     }
 
-
-    // create and initialize the simulation engine
-    // Simulation
     time_t starttime, endtime;
     time(&starttime);
 
     Simulation sim(&config);
 
     if(sim.InitArgs()) {
-        // evacuation time
         double evacTime = 0;
         LOG_INFO("Simulation started with {} pedestrians", sim.GetPedsNumber());
         if(sim.GetAgentSrcManager().GetMaxAgentNumber()) {
-            // Start the thread for managing the sources of agents if any
-            // std::thread t1(sim.GetAgentSrcManager());
             double simMaxTime = config.GetTmax();
             std::thread t1(&AgentsSourcesManager::Run, &sim.GetAgentSrcManager());
-            while(!sim.GetAgentSrcManager().IsRunning()) {
```

Comment:
Any reason why you removed this while?

Reply:
Hi @chraibi good catch /  question! :1st_place_medal: 

I did remove it because "Hey the loop is not doing anything expect spinning w.o. purpose, i.e. no loop body". 

However:
Your review remark promoted me to look into the code and be sure there are no side effects: Turns out there are! :/ The loop is actually blocking the main thread until `AgentSrcManager` has entered its `Run` method. Now the thing is: The loop is probably still useless because although it blocks the main thread until Run has been entered no additional work (that I can see) has been completed before continuing in the main loop. If the intention was here to sync execution of the `AgentSrcManager` with the main sim loop then it is questionable if this works as intended.

In conclusion, I would still remove this loop.

Reply:
Maybe this leads us to another issue, to be opened soon, about "How to force both threads to run nearly at sync?"

Reply:
Actually Id much prefer to just remove the thread completely: See #628